src/Pos/Explorer/Txp/Toil/Logic.hs

```diff
@@ -132,3 +150,59 @@ getTxRelatedAddrs (UnsafeTx {..}, _, _) undo =
     toSet = HS.fromList . toList
     -- Safe here, because union of non-empty sets can't be empty.
     unionNE lhs rhs = NE.fromList $ toList $ HS.union (toSet lhs) (toSet rhs)
+
+combineBalanceUpdates :: BalanceUpdate -> [(Address, (Sign, Coin))]
```

Comment:
Do you mean _return_ instead of _store_? If yes, then we're going to iterate over the result anyway, so there is little comparative advantage. If no, then I don't understand you.